library/core/src/fmt/float.rs

```diff
@@ -228,3 +231,8 @@ macro_rules! floating {
 
 floating! { f32 }
 floating! { f64 }
+
+#[cfg(not(bootstrap))]
+floating! { f16 }
+#[cfg(not(bootstrap))]
+floating! { f128 }
```

Comment:
These methods are not `#[inline]` so they will completely break building the standard library with cg_gcc and cg_clif until these backends support f16 and f128.

Reply:
What is the correct way to fix this? I have also just been using `unimplemented!()` in clif and gcc to make things compiile, I guess those should just throw errors instead?

(down the line of course, I'm still on the very first stage of getting this to work)

Reply:
`#[inline]` functions are only codegened when actually used. So making them `#[inline]` at least temporarily would be an option.

> I guess those should just throw errors instead?

That would replace an ICE with an error when compiling libcore. Both prevent successful compilation.

Reply:
cg_clif and cg_gcc don't yet have CI tests yet right? Could turn them on while I'm working on this if so

Reply:
Try cherry-picking https://github.com/rust-lang/rust/pull/112701 and then remove https://github.com/rust-lang/rust/pull/112701/files#diff-8479eab02701e686aedb15b567dc8fc31220c6e4efb9565ccc9d662b7fee2214R3057-R3063

Reply:
Do these `mini_core` errors usually show up in CI? I don't think I'm getting them on local https://github.com/rust-lang/rust/actions/runs/5806522313/job/15739585197?pr=114607#step:24:3221

Reply:
Because #112701 hasn't landed yet it is possible for PR's in this repo to break cg_clif's testsuite. I fixed that issue just today in https://github.com/bjorn3/rustc_codegen_cranelift/commit/3deb6c69e0ea8665ec4453cfde67491a62683116. Opened https://github.com/rust-lang/rust/pull/114666 to sync it back to this repo.